### ISSUE TITLE
Sending failedError: Sending failed

### ISSUE THREAD

**Issue by ajcollett** [nylas/nylas-mail#3526]
##### Are there any related issues?

#3182
#3331

Seems that there are many that refer to sending issues. 

##### What operating system are you using?
Mac OS X 10.12.4
Ubuntu 17.04

##### What version of Nylas Mail are you using?
Version 2.0.32 (2.0.32)

--

**Bug**
##### Do you have any third-party plugins installed? If so, which ones?
No

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
Office 365 with my work. Gmail works perfectly.

##### Is the issue reproducible with a particular attachment, message, signature, etc?

Every time I try sending an email from the Office 365 address I get the below error. 
If I re-add the account the issue stops for a while then comes back. 
If I clear the email cache, the error is still present. 

Steps:
1. Replay or send an email
2. Get the message:
![screen shot 2017-05-05 at 2 03 14 pm](https://cloud.githubusercontent.com/assets/7753041/25744735/aefec81a-319b-11e7-9509-75d9b2b64bc4.png)
3. Try resend, same result. 

##### Error output
```
Sending failedError: Sending failed
    at NylasError (errors.es6:1:8)
    at APIError (errors.es6:17:5)
    at sendmail-client.es6:53:11
    at throw (<anonymous>)
    at step (sendmail-client.es6:5:1)
    at sendmail-client.es6:5:1
    at tryCatcher (/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/bluebird/js/release/promise.js:687:18)
    at Async._drainQueue (/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/bluebird/js/release/async.js:143:10)
    at Async.drainQueues (/Applications/Nylas Mail.app/Contents/Resources/app.asar/node_modules/bluebird/js/release/async.js:17:14)
```

**Comment by alexisvincent** [comment]
I get the same thing. Have had this issue on and off for months. Also with N1

**Comment by ajcollett** [comment]
Yeah, still happening with me. I'm fairly sure it's a timeout issue. Could you check the ping time between you and office365 please, when this issue happens and when it does not? I ran `ping outlook.office365.com` and got a time of 165ms on average. That doesn't seem too long though...

**Comment by alexisvincent** [comment]
184 on average. But I had the same issue on 2 different google accounts and an office365 account. Have since switched to Airmail

**Comment by alexisvincent** [comment]
@ajcollett Were you at Stellenbosch University?

**Comment by ajcollett** [comment]
Interesting that you asked that. Yeah, I am.

**Comment by alexisvincent** [comment]
Awe, see you on campus. If I'm not mistaken, git errors are redirected to you xD

**Comment by ajcollett** [comment]
He He, yes indeed. Small world. That makes this issue look like it could be and issue in this area. Cheers!
Nylas! How can I give you better feedback!
